Server/Utilities/SpringBoardAlert.m

```diff
@@ -21,17 +21,32 @@ - (instancetype)initWithAlertTitleFragment:(NSString *)alertTitleFragment
 
     self = [super init];
     if (self) {
-        _alertTitleFragment = alertTitleFragment;
+        // replace "%@", "%1$@", "%2$@" by ".+"
+        NSString *tempPattern = [NSString stringWithFormat: alertTitleFragment, @".+", @".+"];
+        // escape string for regex, it is necessary because alert title can contain "+", "*", ".", "?" and etc
+        tempPattern = [NSRegularExpression escapedPatternForString:(NSString *) tempPattern];
+        // apply "^" + string + "$", it needs to match only full alert name
+        // allow to avoid issues when one alert is substring of other alert
+        tempPattern = [NSString stringWithFormat:@"%@%@%@", @"^", tempPattern, @"$"];
+        // fix ".+" after invocation "escapedPatternForString"
+        _alertTitleFragment = [tempPattern stringByReplacingOccurrencesOfString: @"\\.\\+"
+                                                                             withString:@".+"];
         _defaultDismissButtonMarks = dismissButtonTitles;
         _shouldAccept = shouldAccept;
     }
     return self;
 }
 
 - (BOOL)matchesAlertTitle:(NSString *)alertTitle {
-    return [[alertTitle lowercaseString]
-            containsString:[self.alertTitleFragment
-                            lowercaseString]];
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:self.alertTitleFragment
```

Comment:
I think we want to make this regexes persistent - we should create it once and reuse it.

Reply:
Yes, I've got, now I'll try to move regex definition to constructor and test it